src/client.rs

```diff
@@ -52,26 +58,40 @@ impl<T, B> Client<T, B>
 
         debug!("binding client connection");
 
-        let handshake = io::write_all(io, b"PRI * HTTP/2.0\r\n\r\nSM\r\n\r\n")
-            .map_err(ConnectionError::from)
-            .and_then(|(io, _)| {
-                debug!("client connection bound");
-
-                let mut framed_write = proto::framed_write(io);
-                let settings = frame::Settings::default();
-
-                // Send initial settings frame
-                match framed_write.start_send(settings.into()) {
-                    Ok(AsyncSink::Ready) => {
-                        let connection = proto::from_framed_write(framed_write);
-                        Ok(Client { connection })
-                    }
-                    Ok(_) => unreachable!(),
-                    Err(e) => Err(ConnectionError::from(e)),
+        // this is a function so that we can pass a function pointer (rather
```

Comment:
FWIW, [as of Rust 1.19](https://blog.rust-lang.org/2017/07/20/Rust-1.19.html), a closure that doesn't capture over its environment can be cast to a function pointer.

Reply:
Ooh, fancy – I must have missed that change.